### PR TITLE
Documentation developer css

### LINKED ISSUE #25: Document the utility classes in eQ base
Utility classes used in eQ need documenting to avoid duplication or inefficient SCSS.

For example: `u-vh` for visually hidden elements.

### PR DESCRIPTION
**Pull request by armstrongb**
### What is the context of this PR?
The addition of developer / software engineer information for implementation of pattern library outputs and contributions. 

### How to review 
Review the docs added under 'Developer' and add comments or additions to the PR / Branch.

### Issue
Closes #25

### CODE REVIEW

**Comment by armstrongb** [comment]
@saedwards could do with some detail on how the Karma tests are used, implemented and set up (not how to write a Karma test step by step stuff though!)

File: docs/developer/01-implementing-google-analytics.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: Analytics
+label: Google Analytics
+---
+
+### Implementation
+
+By adding the following to elements&hellip;
+
+1. `data-ga="click"`
```
Comment (jrbarnes9) [inline docs/developer/01-implementing-google-analytics.md]: @armstrongb Probably worth adding the 'error' and 'visible' values you can also add to data-ga, as it's not just click events we can track.

File: docs/developer/01-implementing-google-analytics.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  - attaches the Google Analytics click 'event' to this element.
+2. `data-ga-category="accordion"`
+  - sets the component event category recorded by GA.
+3. `data-ga-action="Toggle all panels"`
+  - provides detail on the action this element triggered
+4. `data-ga-label="Show all"`
+  - provides detail on the action this element triggered
+
```
Comment (jrbarnes9) [inline docs/developer/01-implementing-google-analytics.md]: @armstrongb Worth adding an example of the head script devs should use, specifically the setting of 'anonymizeIp', considering its not in the default script.

File: docs/developer/01-implementing-google-analytics.md (hunk 1 above)
Comment (armstrongb) [reply docs/developer/01-implementing-google-analytics.md]: Any others that you can think of?

File: docs/developer/01-implementing-google-analytics.md (hunk 2 above)
Comment (jrbarnes9) [reply docs/developer/01-implementing-google-analytics.md]: LGTM

File: docs/developer/01-implementing-google-analytics.md (hunk 1 above)
Comment (jrbarnes9) [reply docs/developer/01-implementing-google-analytics.md]: No, LGTM

**Comment by boxadesign** [comment]
I think this looks good and is very helpful for devs using the pattern library. 

I'm not sure that the "yarn/gulp" and "karma tests" pages should be part of the pattern library though. I think this kind of instruction should sit in the `readme` in github as that's the point where it's useful i.e. for getting the PL running.

**Comment by armstrongb** [comment]
> I think this looks good and is very helpful for devs using the pattern library.
> 
> I'm not sure that the "yarn/gulp" and "karma tests" pages should be part of the pattern library though. I think this kind of instruction should sit in the `readme` in github as that's the point where it's useful i.e. for getting the PL running.

I'm considering leaving in for now as they form part of the developer related docs but will split out into the wiki or `readme` as they mature.

**Pull request merged**